app/statistics/server/functions/save.js

```diff
@@ -4,6 +4,63 @@ import { statistics } from '../statisticsNamespace';
 statistics.save = function() {
 	const rcStatistics = statistics.get();
 	rcStatistics.createdAt = new Date();
-	Statistics.insert(rcStatistics);
+
+	// Check if there's partial data already saved for the hour
+	const oldStats = Statistics.findLast();
+	if (oldStats) {
+		if (oldStats.connectedUserCountList && oldStats.connectedUserCountList.length) {
+			rcStatistics.avgConnectedUsers = oldStats.connectedUserCountList.reduce((total, value) => total + value) / oldStats.connectedUserCountList.length;
+		} else {
+			rcStatistics.avgConnectedUsers = 0;
+		}
+
+		delete oldStats.connectedUserCountList;
+
+		rcStatistics.totalConnectedUsers = oldStats.totalConnectedUsers || 0;
+		rcStatistics.minConnectedUsers = oldStats.minConnectedUsers;
+		rcStatistics.onlineUsers = oldStats.onlineUsers || 0;
+		rcStatistics.awayUsers = oldStats.awayUsers || 0;
+		rcStatistics.offlineUsers = oldStats.offlineUsers;
+
+		Statistics.update(oldStats._id, rcStatistics);
+	} else {
+		Statistics.insert(rcStatistics);
+	}
+
 	return rcStatistics;
 };
+
+statistics.saveUsersInfo = function() {
+	const userStatistics = statistics.getConnectedUsersStatistics();
+
+	const rcStatistics = Statistics.findLast();
```

Comment:
This will get the latest record, not the record from the day since the partial will be removed only at 3am